### PR TITLE
ci: Upgrade compilers (incl. clang-15)

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -236,8 +236,8 @@ commands:
         type: string
     steps:
       - run:
-          name: "Upgrade codecov"
-          command: sudo pip3 install --upgrade --quiet --no-cache-dir codecov
+          name: "Install codecov"
+          command: pip3 install --break-system-packages --upgrade --user --quiet --no-cache-dir codecov
```
Comment (axic) [inline circle.yml:240]: What is this setting doing?

File: circle.yml [hunk 2]
```diff
@@ -427,7 +427,7 @@ jobs:
     steps:
       - build
       - download_execution_tests:
-          commit: e279c3d0a53e2f3055c7d7f72f993b62dde18e36
+          commit: v11.3
```
Comment (axic) [inline circle.yml:430]: The diff between e279c3d0a53e2f3055c7d7f72f993b62dde18e36 only seems to be concerned with EOF-related test changes.

**Review by axic — APPROVED** [review]
Looks okayish.

File: circle.yml (hunk 1 above)
Comment (chfast) [reply circle.yml:240]: Installs codecov python script. In the new version of pip/Debian it is disallowed to install pip packges "globally". I have not found better way than forcing this with `--break-system-packages`.

File: circle.yml (hunk 2 above)
Comment (chfast) [reply circle.yml:430]: I just picked the latest release which was newer than the previous hash.

File: circle.yml (hunk 1 above)
Comment (axic) [reply circle.yml:240]: > I have not found better way than forcing this with --break-system-packages.

😬

**Review by gumb0 — APPROVED** [review]

**Comment by axic** [comment]
Argh, didn't notice this wasn't on master. Sorry for merging.

**Pull request merged**